commons-db/src/main/java/com/palantir/nexus/db/sql/BasicSQLUtils.java

```diff
@@ -329,20 +328,36 @@ public static void toStringSqlArgs(final StringBuilder sb, Object[] args) {
                         ByteArrayInputStream bais = (ByteArrayInputStream) arg;
                         byte[] bytes = new byte[bais.available()];
                         bais.read(bytes, 0, bais.available());
-                        sb.append(Arrays.toString(bytes));
+                        prettyPrintByteArray(sb, bytes);
+                    } else if (arg instanceof byte[]) {
+                        prettyPrintByteArray(sb, (byte[]) arg);
                     } else {
-                        sb.append(arg.toString());
+                        sb.append(arg);
                     }
                     sb.append("'"); // $NON-NLS-1$
                 }
-                if (i != args.length - 1) {
+                if (i != 0 && i != args.length - 1) {
                     sb.append(", ");
```

Comment:
I think what we had previously was correct - can we add a test please?

Reply:
Done